icu4c/source/i18n/measunit_extra.cpp

```diff
@@ -110,122 +114,72 @@ const struct SIPrefixStrings {
-    "karat",
-    "portion",
-    "bit",
-    "byte",
-    "dot",
-    "pixel",
-    "em",
-    "hertz",
-    "newton",
-    "pound-force",
-    "pascal",
-    "bar",
-    "atmosphere",
-    "ofhg",
-    "electronvolt",
-    "dalton",
-    "joule",
-    "calorie",
-    "british-thermal-unit",
-    "foodcalorie",
-    "therm-us",
-    "watt",
-    "horsepower",
-    "solar-luminosity",
-    "volt",
-    "ohm",
-    "dunam",
-    "acre",
-    "hectare",
-    "teaspoon",
-    "tablespoon",
-    "fluid-ounce-imperial",
-    "fluid-ounce",
-    "cup",
-    "cup-metric",
-    "pint",
-    "pint-metric",
-    "quart",
-    "liter",
-    "gallon",
-    "gallon-imperial",
-    "bushel",
-    "barrel",
-    "knot",
-    "g-force",
-    "lux",
+/**
+ * A ResourceSink that collects table keys from a resource.
+ *
+ * This class is for use by ures_getAllItemsWithFallback. Example code:
+ *
+ *     UErrorCode status = U_ZERO_ERROR;
+ *     const char* unitIdentifiers[200];
+ *     TableKeysSink identifierSink(unitIdentifiers, 200);
+ *     LocalUResourceBundlePointer unitsBundle(ures_openDirect(NULL, "units", &status));
+ *     ures_getAllItemsWithFallback(unitsBundle.getAlias(), "convertUnits", identifierSink, status);
+ */
+class TableKeysSink : public icu::ResourceSink {
+  public:
+    explicit TableKeysSink(const char **out, int32_t outSize)
+        : outArray(out), outSize(outSize), outIndex(0) {
+    }
+
+    /**
+     * Adds the table keys found in value to the output vector.
+     * @param key The key of the resource passed to `value`: the second
+     *     parameter of the ures_getAllItemsWithFallback() call.
+     * @param value Should be a ResourceTable value, if
+     *     ures_getAllItemsWithFallback() was called correctly for this sink.
+     * @param noFallback Ignored.
+     * @param status The standard ICU error code output parameter.
+     */
+    void put(const char * /*key*/, ResourceValue &value, UBool /*noFallback*/, UErrorCode &status) {
+        ResourceTable table = value.getTable(status);
+        if (U_FAILURE(status)) return;
+
+        if (outIndex + table.getSize() > outSize) {
+            status = U_INDEX_OUTOFBOUNDS_ERROR;
+            return;
+        }
+
+        // Collect keys from the table resource.
+        const char *key;
+        for (int32_t i = 0; table.getKeyAndValue(i, key, value); ++i) {
+            U_ASSERT(i < table.getSize());
+            U_ASSERT(outIndex < outSize);
+            outArray[outIndex++] = key;
+        }
+    }
+
+  private:
+    const char **outArray;
+    int32_t outSize;
+    int32_t outIndex;
 };
 
 icu::UInitOnce gUnitExtrasInitOnce = U_INITONCE_INITIALIZER;
 
+// Array of simple unit IDs.
+//
+// The array memory itself is owned by this pointer, but the individual char* in
+// that array point at static memory. (Note that these char* are also returned
+// by SingleUnitImpl::getSimpleUnitID().)
+const char **gSimpleUnits = nullptr;
+
 char *kSerializedUnitExtrasStemTrie = nullptr;
```

Comment:
Optional: this variable should probably have a g suffix instead of k

Reply:
Done. And I went ahead and squashed.